src/plugins/cmakeprojectmanager/cmakebuildconfiguration.h

```diff
@@ -44,14 +44,26 @@ namespace Internal {
 class BuildDirManager;
 class CMakeBuildSettingsWidget;
 
-class CMakeBuildConfiguration : public ProjectExplorer::BuildConfiguration
+class Q_DECL_EXPORT CMakeBuildConfiguration : public ProjectExplorer::BuildConfiguration
 {
     Q_OBJECT
 
     friend class ProjectExplorer::BuildConfigurationFactory;
-    CMakeBuildConfiguration(ProjectExplorer::Target *parent, Core::Id id);
 
 public:
+    CMakeBuildConfiguration(ProjectExplorer::Target *parent, Core::Id id);
+    ~CMakeBuildConfiguration() override;
+
+    QVariantMap toMap() const override;
+    BuildType buildType() const override;
+
+    void initialize(const ProjectExplorer::BuildInfo &info) override;
+    QString disabledReason() const override;
+
+    ProjectExplorer::NamedWidget *createConfigWidget() override;
```

Comment:
Only initialize() and fromMap() are used from MerCMakeBuildConfiguration. I suggest that you make them protected and keep the rest private.